### PR TITLE
add getRelationships use case

### PR DESCRIPTION
**Pull request by mapleleafjack**
## Link to JIRA ticket
[https://hackney.atlassian.net/browse/SCT-202](https://hackney.atlassian.net/browse/SCT-202)

### *What is the problem we're trying to solve*

Expose historic relationships - #28 #27

### *What changes have we introduced*

Added a new use case getRelationships

#### _Checklist_

- [ ] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### *Follow up actions after merging PR*

Create the API endpoint

### CODE REVIEW

File: ResidentsSocialCarePlatformApi.Tests/V1/Helper/TestHelper.cs [hunk 1]
```diff
@@ -167,5 +169,15 @@ public static PersonalRelationship CreatePersonalRelationship(long? personId = n
                 .RuleFor(personalRelationship => personalRelationship.ParentalReponsibility, f => f.Random.String2(1, "YN"))
                 .RuleFor(personalRelationship => personalRelationship.IsInformalCarer, f => f.Random.String2(1, "YN"));
         }
+
+        public static ResidentsSocialCarePlatformApi.V1.Domain.PersonalRelationships CreateRandomPersonalRelationship()
+        {
+            return new Faker<ResidentsSocialCarePlatformApi.V1.Domain.PersonalRelationships>()
+                .RuleFor(personalRelationship => personalRelationship.Parents, f => Enumerable.Range(0, f.Random.Int(0, 5)).Select(x => f.PickRandom(new List<long>() { f.UniqueIndex, f.UniqueIndex, f.UniqueIndex })).ToList())
```
Comment (LBHTKarki) [inline ResidentsSocialCarePlatformApi.Tests/V1/Helper/TestHelper.cs:177]: 👍

File: ResidentsSocialCarePlatformApi.Tests/V1/Helper/TestHelper.cs [hunk 2]
```diff
@@ -167,5 +169,15 @@ public static PersonalRelationship CreatePersonalRelationship(long? personId = n
                 .RuleFor(personalRelationship => personalRelationship.ParentalReponsibility, f => f.Random.String2(1, "YN"))
                 .RuleFor(personalRelationship => personalRelationship.IsInformalCarer, f => f.Random.String2(1, "YN"));
         }
+
+        public static ResidentsSocialCarePlatformApi.V1.Domain.PersonalRelationships CreateRandomPersonalRelationship()
```
Comment (LBHTKarki) [inline ResidentsSocialCarePlatformApi.Tests/V1/Helper/TestHelper.cs]: Would it be easier to add using for ResidentsSocialCarePlatformApi.V1.Domain? Faster to type and easier to read if using only the class name

File: ResidentsSocialCarePlatformApi/V1/Domain/PersonalRelationships.cs [hunk 3]
```diff
@@ -4,6 +4,14 @@ namespace ResidentsSocialCarePlatformApi.V1.Domain
...
     {
+        public PersonalRelationships()
+        {
+            Parents = new List<long>();
+            Siblings = new List<long>();
+            Children = new List<long>();
+            Other = new List<long>();
+        }
+
         public List<long> Parents { get; set; }
```
Comment (LBHTKarki) [inline ResidentsSocialCarePlatformApi/V1/Domain/PersonalRelationships.cs:15]: `public List<long> Parents { get; set; } = new List<long>(); ` syntax could be used here as well. Using constructor like that gives the same end result of course.

File: ResidentsSocialCarePlatformApi/V1/UseCase/GetRelationshipsByPersonIdUseCase.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace ResidentsSocialCarePlatformApi.V1.UseCase
+{
+    public class GetRelationshipsByPersonIdUseCase : IGetRelationshipsByPersonIdUseCase
+    {
+        ISocialCareGateway _socialCareGateway;
+
+        public GetRelationshipsByPersonIdUseCase(ISocialCareGateway socialCareGateways)
+        {
+            this._socialCareGateway = socialCareGateways;
```
Comment (LBHTKarki) [inline ResidentsSocialCarePlatformApi/V1/UseCase/GetRelationshipsByPersonIdUseCase.cs]: `this` keyword is not necessary here

File: ResidentsSocialCarePlatformApi.Tests/V1/Helper/TestHelper.cs (hunk 2 above)
Comment (mapleleafjack) [reply ResidentsSocialCarePlatformApi.Tests/V1/Helper/TestHelper.cs]: Added :)

File: ResidentsSocialCarePlatformApi/V1/Domain/PersonalRelationships.cs (hunk 3 above)
Comment (mapleleafjack) [reply ResidentsSocialCarePlatformApi/V1/Domain/PersonalRelationships.cs:15]: I actually wanted to keep them separate, as I like to initialise them in the constructor.
Let me know if you prefer the other syntax, I can change it :)

File: ResidentsSocialCarePlatformApi/V1/UseCase/GetRelationshipsByPersonIdUseCase.cs (hunk 4 above)
Comment (mapleleafjack) [reply ResidentsSocialCarePlatformApi/V1/UseCase/GetRelationshipsByPersonIdUseCase.cs]: Updated :)

File: ResidentsSocialCarePlatformApi/V1/Domain/PersonalRelationships.cs (hunk 3 above)
Comment (LBHTKarki) [reply ResidentsSocialCarePlatformApi/V1/Domain/PersonalRelationships.cs:15]: That's absolutely fine. You can use whichever syntax you prefer

**Pull request merged**